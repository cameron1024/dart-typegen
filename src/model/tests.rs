use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
