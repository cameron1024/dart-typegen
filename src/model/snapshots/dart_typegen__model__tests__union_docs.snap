---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    defaults: None,
    enums: [],
    classes: [],
    unions: [
        Union {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Animal",
                span: SourceSpan {
                    offset: SourceOffset(
                        48,
                    ),
                    length: 8,
                },
            },
            sealed: None,
            json_discriminant: None,
            json_discriminant_value_case: None,
            docs: Some(
                SpannedScalar {
                    value: "\nSome docs \n\nSome more lines\n  ",
                    span: SourceSpan {
                        offset: SourceOffset(
                            66,
                        ),
                        length: 36,
                    },
                },
            ),
            classes: [
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Dog",
                        span: SourceSpan {
                            offset: SourceOffset(
                                111,
                            ),
                            length: 5,
                        },
                    },
                    fields: [
                        Field {
                            name: SpannedScalar {
                                value: "name",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        129,
                                    ),
                                    length: 6,
                                },
                            },
                            ty: SpannedScalar {
                                value: "String",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        141,
                                    ),
                                    length: 8,
                                },
                            },
                            defaults_to: None,
                            defaults_to_dart: None,
                            docs: None,
                            to_json: None,
                            from_json: None,
                            json_key: None,
                        },
                    ],
                    docs: None,
                    json_key_case: None,
                    json_discriminant_value: None,
                    extra_dart: [],
                },
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Cat",
                        span: SourceSpan {
                            offset: SourceOffset(
                                163,
                            ),
                            length: 5,
                        },
                    },
                    fields: [
                        Field {
                            name: SpannedScalar {
                                value: "age",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        181,
                                    ),
                                    length: 5,
                                },
                            },
                            ty: SpannedScalar {
                                value: "int",
                                span: SourceSpan {
                                    offset: SourceOffset(
                                        192,
                                    ),
                                    length: 5,
                                },
                            },
                            defaults_to: None,
                            defaults_to_dart: None,
                            docs: None,
                            to_json: None,
                            from_json: None,
                            json_key: None,
                        },
                    ],
                    docs: None,
                    json_key_case: None,
                    json_discriminant_value: None,
                    extra_dart: [],
                },
            ],
            extra_dart: [],
        },
    ],
}
